enhancements/observability-ui/observability-ui-operator.md

```diff
@@ -0,0 +1,265 @@
+---
+title: observability-ui-operator
+authors:
+  - "@jgbernalp"
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - "@kyoto"
+  - "@zhuje"
+  - "@alanconway"
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - "@stleerh"
+  - "@eparis"
+api-approvers:
+  - "@spadgett"
+creation-date: 2023-09-18
+last-updated: 2023-10-11
+tracking-link:
+  - https://issues.redhat.com/browse/OU-204
+see-also:
+  - ""
+replaces:
+  - ""
+superseded-by:
+  - ""
+---
+
+title: observability-ui-operator
+
+# Observability UI Operator
+
+The Observability UI Operator aims to manage dynamic console plugins for observability signals inside the OpenShift console, ensuring a consistent user experience and efficient management of UI plugins.
+
+## Summary
+
+This proposal introduces the Observability UI Operator, a tool designed to manage UI plugins related to observability signals within the OpenShift console. By centralizing the management of such plugins, we can offer a unified observability experience in the console and accommodate new use cases, all while decoupling UI responsibilities from other operators.
+
+## Motivation
+
+### Why:
+
+The current state of observability signals in the OpenShift console has each operator responsible for its own console plugin. This sometimes results in operators deploying plugins outside their primary scope. As the requirements for the console's UI grow, there's a clear need for a centralized system that can manage diverse UI components spanning across various signals to offer a unified observability experience in the console.
+
+### User Stories
+
+- As an OpenShift user, I want an operator from the Red Hat catalog that can deploy various observability UI components so that all signals supported by the cluster are easily accessible and can be used for troubleshooting.
+
+- As an OpenShift administrator, I want a centralized operator for observability UI components so that I can streamline console requirements and integrate diverse signals effectively.
```

Comment:
A unique console plugin for all observability operators would solve that point without the need to install an extra operator. As soon as you have at least one installed, the observability plugin is there.
Behind the scene, the plugin can feature gate the exposed pages according to available metrics / logs etc.

The challenge in this approach would be the compatibility between plugin version and each operator version but there are ways to remediate that.

Reply:
> A unique console plugin for all observability operators would solve that point without the need to install an extra operator. As soon as you have at least one installed, the observability plugin is there.

@jpinsonneau But how does the observability plugin get installed? Is the suggestion for all observability operators to include a copy of the plugin?

Reply:
Yes, all operators could embed the same plugin and the most up to date is taken. It need to create a mechanism to identify who's responsible of it, using owner / version labels for example. The code managing that must be shared between all operators to avoid any reconcile issue. It's a cross team effort !

As an alternative, the plugin could be embedded in Monitoring Operator or OCP Console directly, and enabled when needed. The downside of this approach is the update cycle, tied to the owner.

Both of these approach seems to be a better match for the [objectives](https://github.com/jgbernalp/openshift-enhancements/blob/observability-ui-operator/enhancements/observability-ui/observability-ui-operator.md#objective) listed in this doc rather than creating yet another operator.

Reply:
I agree, and the simplest solution is still having this shared piece into Cluster Monitoring Operator or OCP Console out of the box.

What are the reasons not doing that (appart from the update cycle we won't control) ?

Reply:
@jpinsonneau the main problem this aims to solve is the update cycle by being outside the OCP console or monitoring operator to match the signal operators cycles. Another reason, is having operators to deploy a plugin they don't even consume. For example LokiOperator does not deploy the logging plugin as the operator is not specific to OpenShift, so the cluster logging operator deploys it even though the plugin never consumes services from the logging operator.

Reply:
@stleerh I believe this will give customers the flexibility to add visualization based on their needs, if in some case metrics or logs are just forwarded, the visualization piece might not be needed saving resources. for multi-cluster setups it should be enabled only in the hub cluster . If visualization should be installed by default this can come from a meta operator like OBO that enables the signal operator and enables that piece in the Observability UI operator.

Reply:
So to me it's still a bit heavy to create a dedicated operator, not including monitoring, just to avoid update cycles.

It would make more sense if the target was:
- including monitoring dashboards / metrics query pages
- sharing storages (Prometheus, Loki etc)
- managing gateway + roles
- enabling correlation between any metrics / storages

Your plugin should rely on the `logStore` you configured in `ClusterLogging` CR, so even if it doesn't consume the logging services, it rely on its configuration.

Our Console Plugin is deployed by Network Observability Operator with Loki storage behind the scene, just as same as you:
- both `collector` and `plugin` clients for Loki are configured [in the same section of FlowCollector CR](https://github.com/netobserv/network-observability-operator/blob/release-1.4/api/v1beta1/flowcollector_types.go#L61-L62)
- [Plugin is optionnal](https://github.com/netobserv/network-observability-operator/blob/release-1.4/pkg/helper/flowcollector.go#L105)
- FYI, even Loki [became optionnal](https://github.com/netobserv/network-observability-operator/blob/release-1.4/api/v1beta1/flowcollector_types.go#L521-L523) since we want to provide a lightweight observability based on Prometheus metrics

Reply:
The correlation is given by korrel8r and enhanced customizable dashboards by Perses. So the observability operator is intended to do this heavy lifting of configuring and installing the Perses operator and korrel8r for correlation.

The logging plugin could be connected to any Loki store without even having cluster logging in a cluster, and the only task for the logging operator is to enable it. So it currently relies on the CR configuration but this is not ideal as any other operator can do the job. 

IIUC even if the net observe plugin is optional is coupled to the net observe operator backend, regardless of the store. This is a reason why we wont include the network observability plugin as it has a clear 1:1 relationship with its operator.

Reply:
In the current implementation yes, however that 1:1 relationship may change in the future, depending on multi cluster needs.
I think that's the reason why logging plugin can be installed without the operator right ?

Reply:
yes this will be the ideal in multi cluster scenario so visualization is present only where is needed.